listenbrainz/tests/integration/test_playlist_api.py

```diff
@@ -212,6 +212,35 @@ def test_playlist_create_empty(self):
         # Make sure the return playlist id is valid
         UUID(response.json["playlist_mbid"])
 
+    def test_playlist_xspf_additional_metadata(self):
+        """ Test for checking that additional meta data field is properly constructed and not causing a crash """
+
+        playlist = {
+            "playlist": {
+                "title": "you're a person",
+                "extension": {
+                    PLAYLIST_EXTENSION_URI: {
+                        "public": True,
+                    }
+                },
+                "additional_metadata": {
+                    "note": "a great playlist",
+                    "details": {
+                         "mood": "varied",
+                         "genre": "eclectic",
+                         "favorite_track": "Random Song"
+                        }
+                    }
+                }
+            }
+        
+        response = self.client.post(
+            url_for("playlist_api_v1.create_playlist"),
+            json=playlist,
+            headers={"Authorization": "Token {}".format(self.user["auth_token"])}
+        )
+        self.assert200(response)
```

Comment:
You also need to make a call to the serialize_xspf endpoint after this, and ensure that it responds with a 200 response too.

Reply:
Thanks for pointing that out